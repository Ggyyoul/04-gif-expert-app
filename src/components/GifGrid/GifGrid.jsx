export const GifGrid = ({category}) => {

  const getGifs = async () => {
    const url = 'http://api.giphy.com/v1/gifs/search?api_key=oBzdrIoXRePjP71ync4RXn0SxeThaJKc&q=pokemon&limit=5'
    const resp = await fetch(url)
    const {data}   = await resp.json()
    
    const gifs = data.map(data => {
      return {
        id:data.id,
        image: data.images.downsized.url,
        title: data.title

      }
    })

  }
  getGifs()

  return (
    <div>
      <h3>{category}</h3>
    </div>
  )
}
