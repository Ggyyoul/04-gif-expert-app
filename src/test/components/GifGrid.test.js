import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test in component  <GifGrid />', () => {
  const CATEGORY = 'Twice';

  test('should have match with the snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    let wrapper = shallow(<GifGrid category={CATEGORY} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show img with useFetchGifs is loaded', () => {
    const gifs = [
      {
        id: 'ABB',
        url: 'https://localhostproeba/cosa.jpg',
        title: 'cualquier cosa',
      },
      {
        id: 'ABC',
        url: 'https://localhostproeba/cosa.jpg',
        title: 'cualquier cosa',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={CATEGORY} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });

});
