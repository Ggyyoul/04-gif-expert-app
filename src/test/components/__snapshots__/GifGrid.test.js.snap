// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component  <GifGrid /> should have match with the snapshot 1`] = `
<Fragment>
  <h3>
    Twice
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test in component  <GifGrid /> should show img with useFetchGifs is loaded 1`] = `
<Fragment>
  <h3>
    Twice
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABB"
      key="ABB"
      title="cualquier cosa"
      url="https://localhostproeba/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhostproeba/cosa.jpg"
    />
  </div>
</Fragment>
`;
